<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.555">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Paul Bürkner - Research Projects</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="../site_libs/quarto-contrib/fontawesome6-0.1.0/all.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/fontawesome6-0.1.0/latex-fontsize.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Paul Bürkner</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../research/index.html"> 
<span class="menu-text">Research</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../people/index.html"> 
<span class="menu-text">People</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../projects/index.html" aria-current="page"> 
<span class="menu-text">Projects</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../publications/index.html"> 
<span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../software/index.html"> 
<span class="menu-text">Software</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../talks/index.html"> 
<span class="menu-text">Talks</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../theses/index.html"> 
<span class="menu-text">Theses</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../positions/index.html"> 
<span class="menu-text">Positions</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../cv/index.html"> 
<span class="menu-text">CV</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="mailto:paul.buerkner@gmail.com"> 
<span class="menu-text"><i class="fa-solid fa-envelope" aria-label="envelope"></i></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/paul-buerkner"> <i class="bi bi-github" role="img" aria-label="github">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://scholar.google.com/citations?user=JSj6m1IAAAAJ&amp;hl"> 
<span class="menu-text"><i class="fa-solid fa-graduation-cap" aria-label="graduation-cap"></i></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/paulbuerkner"> <i class="bi bi-twitter" role="img" aria-label="twitter">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
        
    </div>
<!-- main -->
<main class="content column-page" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Research Projects</h1>
</div>



<div class="quarto-title-meta column-page">

    
  
    
  </div>
  


</header>


<section id="bayesflow-software" class="level2">
<h2 class="anchored" data-anchor-id="bayesflow-software">BayesFlow: Simulation Intelligence with Deep Learning</h2>
<p>Simulation intelligence (SI) subsumes an emerging generation of scientific methods which utilize digital simulations for emulating and understanding complex real-world systems and phenomena. Recently, neural networks and deep learning have demonstrated a great potential for accelerating and scaling up SI to previously intractable problems and data sets. However, the availability of user-friendly software is still limited, which hampers the widespread and flexible use of modern SI methods.</p>
<p>In this project, we focus on software for amortized Bayesian inference, which is an essential part of SI. The hallmark feature of amortized Bayesian inference is an upfront training phase (e.g., of a neural network), which is then amortized by a nearly instant fully Bayesian inference for an arbitrary number of data sets during test time. Concretely, we aim to advance the <a href="../software#bayesflow">BayesFlow research software library</a> into becoming the long-term, gold-standard software for amortized Bayesian inference.</p>
<p>Overarching Topics: <a href="../research#abi" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Amortized Inference</a> <a href="../research#machine-workflow" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Machine-Assisted Workflows</a> <a href="../research#uncertainty-quantification" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Uncertainty Quantification</a></p>
<p>Project Members: <a href="../people#lars-kuehmichel" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Lars Kühmichel</a></p>
<p>Funders: <a href="https://www.dfg.de/en/index.jsp" class="btn btn-outline-primary btn-page-header btn-xs" role="button">German Research Foundation (DFG)</a></p>
<p>Funding Period: 2024 – 2027</p>
<p>Publications:</p>
<ul>
<li>Habermann D., Schmitt M., Kühmichel L., Bulling A., Radev S. T., &amp; Bürkner P. C. (in review). Amortized Bayesian Multilevel Models. <em>ArXiv preprint</em>. <a href="https://arxiv.org/abs/2408.13230" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a></li>
</ul>
</section>
<section id="abi-applications" class="level2">
<h2 class="anchored" data-anchor-id="abi-applications">Applications of Amortized Bayesian Inference</h2>
<p>Recent developments in simulation-based amortized inference have ushered in new possibilities for conducting principled Bayesian analysis. The simulation-based approach unlocks the potential of complex models whose likelihoods or priors are not analytically tractable. Amortized approaches make the required computations relatively fast, thus allowing for the deployment of intricate models in scenarios that were hitherto deemed unfeasible or inconvenient. Nevertheless, the novelty of this approach poses a challenge, as its widespread adoption hinges on the availability of user-friendly documentation and resources that simplify entry into the field, as well as empirical examples that validate the method’s usefulness for the practical researchers.</p>
<p>In this project, our emphasis is on applications within cognitive modeling and developmental psychology. We focus on how simulation-based amortized inference can address important challenges within the field, not only during the data analysis phase but also in the planning and execution of studies and experiments. As a by-product we will generate tutorials and educational materials providing gentle introductions into the topic. This project also aims to lay the foundations for integrating simulation-based amortized inference with popular statistical software packages used by practitioners who may not have extensive coding skills, thereby broadening the scope of users benefiting from its advantages.</p>
<p>Overarching Topics: <a href="../research#abi" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Amortized Inference</a> <a href="../research#lvm" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Latent Variable Modeling</a> <a href="../research#uncertainty-quantification" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Uncertainty Quantification</a></p>
<p>Project Members: <a href="../people#simon-kucharsky" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Šimon Kucharský</a></p>
<p>Funders: <a href="https://www.tu-dortmund.de/en/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">TU Dortmund University</a></p>
<p>Funding Period: 2024 – 2027</p>
</section>
<section id="amortized-mlms" class="level2">
<h2 class="anchored" data-anchor-id="amortized-mlms">Amortized Bayesian Inference for Multilevel Models</h2>
<div class="quarto-figure quarto-figure-left">
<figure class="figure">
<p><img src="../images/mlm_overview.png" class="img-fluid quarto-figure quarto-figure-left figure-img" alt="An illustration of amortized multilevel models."></p>
</figure>
</div>
<p>Probabilistic multilevel models (MLMs) are a central building block in Bayesian data analysis. Despite their widely acknowledged advantages, MLMs remain challenging to estimate and evaluate, especially when the involved likelihoods or priors are analytically intractable. Recent developments in generative deep learning and simulation-based inference have shown promising results in scaling up Bayesian inference through amortization. However, the utility of deep generative models for learning Bayesian MLMs remains largely unexplored.</p>
<p>In this project, we propose to develop a general and efficient neural inference framework for estimating and evaluating complex Bayesian MLMs. Our framework will substantially extend previous work on simulation-based Bayesian inference for single-level models. Moreover, it aims to encompass not only the inference phase of a Bayesian workflow but also the model evaluation steps, which usually comprise a computational bottleneck with standard (non-amortized) Bayesian methods. Thus, the proposed project has the potential to greatly enhance model-based inference and understanding of complex processes across the quantitative sciences.</p>
<p>Overarching Topics: <a href="../research#abi" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Amortized Inference</a> <a href="../research#lvm" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Latent Variable Modeling</a> <a href="../research#uncertainty-quantification" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Uncertainty Quantification</a></p>
<p>Project Members: <a href="../people#daniel-habermann" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Daniel Habermann</a></p>
<p>Funders: <a href="https://www.dfg.de/en/index.jsp" class="btn btn-outline-primary btn-page-header btn-xs" role="button">German Research Foundation (DFG)</a></p>
<p>Funding Period: 2023 – 2026</p>
<p>Publications:</p>
<ul>
<li><p>Habermann D., Schmitt M., Kühmichel L., Bulling A., Radev S. T., &amp; Bürkner P. C. (in review). Amortized Bayesian Multilevel Models. <em>ArXiv preprint</em>. <a href="https://arxiv.org/abs/2408.13230" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a></p></li>
<li><p>Schmitt M., Ivanova D. R., Habermann D., Köthe U., Bürkner P. C., &amp; Radev S. T. (accepted). Leveraging Self-Consistency for Data-Efficient Amortized Bayesian Inference. <em>Proceedings of the International Conference on Machine Learning (ICML)</em>. <a href="https://arxiv.org/abs/2310.04395" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a></p></li>
</ul>
<!-- -->
</section>
<section id="bdlvms" class="level2">
<h2 class="anchored" data-anchor-id="bdlvms">Bayesian Distributional Latent Variable Models</h2>
<div class="quarto-figure quarto-figure-left">
<figure class="figure">
<p><img src="../images/distributional_SEMs.png" class="img-fluid quarto-figure quarto-figure-left figure-img" alt="An illustration of distributional SEMs."></p>
</figure>
</div>
<p>In psychology and related sciences, a lot of research is concerned with studying latent variables, that is, constructs which are not directly observable. Statistical methods for modeling latent variables based on manifest (observable) indicators are thus crucial to the scientific progress in those fields. Two major interconnected statistical areas dealing with latent variables exist, namely, Item Response Theory (IRT) and Structural Equation Modeling (SEM). Although the two fields are closely connected, the frontiers of IRT and SEM have developed in somewhat different directions.</p>
<p>A combination of these two major frontiers would enable researchers to tackle a lot of advanced psychological research questions at the intersection of psychometrics, personnel psychology, cognitive psychology, and applied psychology. In order for us to gain better insights into behavioral and cognitive processes, their mathematical approximations should match the processes’ complexity in both overall distributional form and its components that are expressed as complex functions of predicting variables.</p>
<p>This project aims to develop a framework for Bayesian distributional latent variable models that combines the principles of IRT and SEM with the flexibility of distributional regression powered by modern Bayesian estimation methods.</p>
<p>Overarching Topics: <a href="../research#lvm" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Latent Variable Modeling</a> <a href="../research#machine-workflow" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Machine-Assisted Workflows</a> <a href="../research#uncertainty-quantification" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Uncertainty Quantification</a></p>
<p>Project Members: <a href="../people#luna-fazio" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Luna Fazio</a></p>
<p>Funders: <a href="https://www.dfg.de/en/index.jsp" class="btn btn-outline-primary btn-page-header btn-xs" role="button">German Research Foundation (DFG)</a> <a href="https://www.tu-dortmund.de/en/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">TU Dortmund University</a></p>
<p>Funding Period: 2022 – 2025</p>
<p>Publications:</p>
<ul>
<li><p>Fazio L., Scholz M., &amp; Bürkner P. C. (in review). Generative Bayesian Modeling with Implicit Priors. <em>ArXiv preprint</em>. <a href="https://arxiv.org/abs/2408.06504" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://github.com/sims1253/implicit-priors" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Code &amp; Data</a></p></li>
<li><p>Fazio L. &amp; Bürkner P. C. (in review). Gaussian distributional structural equation models: A framework for modeling latent heteroscedasticity. <em>ArXiv preprint</em>. <a href="https://arxiv.org/abs/2404.14124" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://github.com/bdlvm-project/gdsem-paper" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Code &amp; Data</a></p></li>
</ul>
<!-- -->
</section>
<section id="sbpriors" class="level2">
<h2 class="anchored" data-anchor-id="sbpriors">Simulation-Based Prior Distributions for Bayesian Models</h2>
<div class="quarto-figure quarto-figure-left">
<figure class="figure">
<p><img src="../images/simulation-based-priors.png" class="img-fluid quarto-figure quarto-figure-left figure-img" alt="An illustration of the simulation-based prior workflow."></p>
</figure>
</div>
<p>Data-driven statistical modeling plays a crucial role in almost all quantitative sciences. Despite continuous increases in the amount of available data, the addition of further information sources, such as expert knowledge, often remains an irreplaceable part of setting up high-fidelity models. Grounded in probability theory, Bayesian statistics provides a principled approach to including expert knowledge in the form of prior distributions, a process called prior elicitation. However, prior elicitation for high-dimensional Bayesian models is infeasible with existing methods due to practical and computational challenges. With the goal of solving these challenges, we propose to develop simulation-based priors for high-dimensional Bayesian models that allow to incorporate prior information elicited on any model-implied quantities. We expect the developed methods to have a major impact on all fields applying probabilistic modeling by making the use of expert knowledge practical, robust, and computationally feasible.</p>
<p>Overarching Topics: <a href="../research#prior-specification" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Prior Specification</a> <a href="../research#abi" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Amortized Inference</a> <a href="../research#uncertainty-quantification" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Uncertainty Quantification</a></p>
<p>Project Members: <a href="../people#florence-bockting" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Florence Bockting</a></p>
<p>Funders: <a href="https://www.tu-dortmund.de/en/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">TU Dortmund University</a> <a href="https://www.simtech.uni-stuttgart.de/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Cluster of Excellence SimTech</a></p>
<p>Funding Period: 2022 – 2025</p>
<p>Publications:</p>
<ul>
<li>Bockting F., Radev S. T., &amp; Bürkner P. C. (2024). Simulation-Based Prior Knowledge Elicitation for Parametric Bayesian Models. <em>Scientific Reports</em>. doi:10.1038/s41598-024-68090-7 <a href="../publications/pdf/2024__Bockting_et_al__Scientific_Reports.pdf" class="btn btn-outline-primary btn-page-header btn-xs" role="button">PDF</a> <a href="https://www.nature.com/articles/s41598-024-68090-7" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Journal</a> <a href="http://arxiv.org/abs/2308.11672" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://florence-bockting.github.io/PriorLearning/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Website</a> <a href="https://github.com/florence-bockting/PriorLearning" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Code &amp; Data</a></li>
</ul>
<!-- -->
</section>
<section id="scrna-models" class="level2">
<h2 class="anchored" data-anchor-id="scrna-models">Probabilistic Models for Single-Cell RNA Sequencing Data</h2>
<div class="quarto-figure quarto-figure-left">
<figure class="figure">
<p><img src="../images/latent-derivative-gps.png" class="img-fluid quarto-figure quarto-figure-left figure-img" alt="An illustration of our latent derivative Gaussian process framework."></p>
</figure>
</div>
<p>Trajectory and pseudo-time inference methods in single-cell RNA sequencing face challenges from the ambiguity of the static single-cell transcriptome snapshot data. In this project, we aim to tackle this challenge by means of advanced probabilistic methods. Concretely, we aim to reconstruct unobserved cell ordering as latent pseudo-time by analyzing RNA spliced counts and corresponding derivative RNA velocity. Further, we aim to obtain uncertainty estimates of the latent cell ordering using Bayesian inference. To achieve these goals, we will develop advanced latent Gaussian process models with the ability of utilizing derivative information to increase precision in estimating unobserved latent inputs. This model deploys derivative covariance kernel functions and modifications in the hyperparameter specifications, thus increasing capabilities for utilizing derivative information in a multi-output setting. Although the primary motivation lies in applications in single-cell biology, this model has the potential to solve similar research problems dealing with multi-output data and its derivatives from diverse fields of study.</p>
<p>Overarching Topics: <a href="../research#lvm" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Latent Variable Modeling</a> <a href="../research#uncertainty-quantification" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Uncertainty Quantification</a></p>
<p>Project Members: <a href="../people#soham-mukherjee" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Soham Mukherjee</a></p>
<p>Co-Supervisors: <a href="https://claassenlab.github.io/people/manfred-claassen/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Manfred Claassen</a></p>
<p>Funders: <a href="https://www.tu-dortmund.de/en/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">TU Dortmund University</a> <a href="https://uni-tuebingen.de/en/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">University of Tübingen</a></p>
<p>Funding Period: 2022 – 2025</p>
<p>Publications:</p>
<ul>
<li>Mukherjee S., Claassen M., Bürkner P. C. (in review). DGP-LVM: Derivative Gaussian process latent variable model. <em>ArXiv preprint</em>. <a href="https://arxiv.org/abs/2404.04074" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://github.com/Soham6298/DGP-LVM" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Code &amp; Data</a></li>
</ul>
<!-- -->
</section>
<section id="uncertainty-surrogate-models" class="level2">
<h2 class="anchored" data-anchor-id="uncertainty-surrogate-models">Data-Integrated Training of Surrogate Models for Uncertainty Quantification</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/uncertainty_prop_workflow.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:80.0%" alt="Surrogate uncertainty propagation workflow."></p>
</figure>
</div>
<p>Uncertainty quantification is crucial to assess the predictive power and limitations of complex systems models. However, in the case of high-dimensional parameter spaces and/or complex functional relationships, physics-based simulation models are often computationally too demanding for rigorous Bayesian uncertainty quantification. Surrogate models allow for such analyses with much lower effort. They are typically trained such that they fit the simulation reference best.</p>
<p>What is left unexplored is the possibility of surrogate models to actually fit observed data better than the reference model. This phenomenon occurs when structural misspecification of the physics-constrained reference model limits its performance, but at the same time, the more flexible data-driven surrogate model can better capture the relation of output and input data. Such situations offer huge potential for diagnostic evaluation of the modelling approach toward deeper system understanding and model improvement.</p>
<p>We aim at developing (1) a weighted data-integrated surrogate training approach for improved predictive performance, (2) a diagnostic approach for structural error detection in the reference model, and (3) an uncertainty propagation analysis that accounts for the approximation error introduced by the use of surrogates.</p>
<p>Overarching Topics: <a href="../research#uncertainty-quantification" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Uncertainty Quantification</a></p>
<p>Project Members: <a href="../people#philipp-reiser" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Philipp Reiser</a></p>
<p>Co-Supervisors: <a href="https://www.simtech.uni-stuttgart.de/exc/people/Guthke/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Anneli Guthke</a></p>
<p>Funders: <a href="https://www.simtech.uni-stuttgart.de/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Cluster of Excellence SimTech</a></p>
<p>Funding Period: 2022 – 2025</p>
<p>Publications:</p>
<ul>
<li>Reiser P., Aguilar J. E., Guthke A., &amp; Bürkner P. C. (in review). Uncertainty Quantification and Propagation in Surrogate-based Bayesian Inference. <em>ArXiv preprint</em>. <a href="https://arxiv.org/abs/2312.05153" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://github.com/philippreiser/bayesian-surrogate-uncertainty-paper" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Code &amp; Data</a></li>
</ul>
<!-- -->
</section>
<section id="mu-bmb" class="level2">
<h2 class="anchored" data-anchor-id="mu-bmb">Meta-Uncertainty in Bayesian Model Comparison</h2>
<p><img src="../images/meta_uncertainty_banner.png" class="img-fluid" alt="An illustration of meta-uncertainty."></p>
<p>What we can learn from a single data set in experiments and observational studies is always limited, and we are inevitably left with some remaining uncertainty. It is of utmost importance to take this uncertainty into account when drawing conclusions if we want to make real scientific progress. Formalizing and quantifying uncertainty is thus at the heart of statistical methods aiming to obtain insights from data.</p>
<p>To compare scientific theories, scientists translate them into statistical models and then investigate how well the models’ predictions match the gathered real-world data. One widely applied approach to compare statistical models is Bayesian model comparison (BMC). Relying on BMC, researchers obtain the probability that each of the competing models is true (or is closest to the truth) given the data. These probabilities are measures of uncertainty and, yet, are also uncertain themselves. This is what we call meta-uncertainty (uncertainty over uncertainties). Meta-uncertainty affects the conclusions we can draw from model comparisons and, consequently, the conclusions we can draw about the underlying scientific theories.</p>
<p>This project contributes to this endeavor by developing and evaluating methods for quantifying meta-uncertainty in BMC. Building upon mathematical theory of meta-uncertainty, we will utilize extensive model simulations as an additional source of information, which enable us to quantify so-far implicit yet important assumptions of BMC. What is more, we will be able to differentiate between a closed world, where the true model is assumed to be within the set of considered models, and an open world, where the true model may not be within that set – a critical distinction in the context of model comparison procedures.</p>
<p>Overarching Topics: <a href="../research#model-comparison" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Model Comparison</a> <a href="../research#uncertainty-quantification" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Uncertainty Quantification</a> <a href="../research#abi" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Amortized Inference</a></p>
<p>Project Members: <a href="../people#marvin-schmitt" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Marvin Schmitt</a></p>
<p>Funders: <a href="https://cyber-valley.de/en/research-fund" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Cyber Valley Research Fund</a></p>
<p>Funding Period: 2021 – 2024</p>
<p>Publications:</p>
<ul>
<li><p>Schmitt M., Radev S. T., &amp; Bürkner P. C. (in review). Fuse It or Lose It: Deep Fusion for Multimodal Simulation-Based Inference. <em>ArXiv preprint</em>. <a href="https://arxiv.org/abs/2311.10671" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a></p></li>
<li><p>Schmitt M., Pratz V., Köthe U., Bürkner P. C., &amp; Radev S. T. (in review). Consistency Models for Scalable and Fast Simulation-Based Inference. <em>ArXiv preprint</em>. <a href="https://arxiv.org/abs/2312.05440" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a></p></li>
<li><p>Schmitt M., Hikida Y., Radev S. T., Sadlo F., &amp; Bürkner P. C. (in review). The Simplex Projection: Lossless Visualization of 4D Compositional Data on a 2D Canvas. <em>ArXiv preprint</em>. <a href="https://arxiv.org/abs/2403.11141" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://github.com/marvinschmitt/ggsimplex" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Software</a> <a href="https://www.marvinschmitt.com/blog/ggsimplex-prerelease/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Tutorial</a></p></li>
<li><p>Schmitt M., Ivanova D. R., Habermann D., Köthe U., Bürkner P. C., &amp; Radev S. T. (accepted). Leveraging Self-Consistency for Data-Efficient Amortized Bayesian Inference. <em>Proceedings of the International Conference on Machine Learning (ICML)</em>. <a href="https://arxiv.org/abs/2310.04395" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a></p></li>
<li><p>Elsemüller L., Olischläger H., Schmitt M., Bürkner P. C., Köthe U., &amp; Radev S. T. (2024). Sensitivity-Aware Amortized Bayesian Inference. <em>Transactions in Machine Learning Research</em>. <a href="../publications/pdf/2024__Elsemueller_et_al__TMLR.pdf" class="btn btn-outline-primary btn-page-header btn-xs" role="button">PDF</a> <a href="https://openreview.net/forum?id=Kxtpa9rvM0" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Journal</a> <a href="https://arxiv.org/abs/2310.11122" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://github.com/bayesflow-org/SA-ABI" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Code &amp; Data</a></p></li>
<li><p>Schmitt, M., Radev, S. T., &amp; Bürkner P. C. (2023). Meta-Uncertainty in Bayesian Model Comparison. <em>Artificial Intelligence and Statistics (AISTATS) Conference Proceedings</em>. <a href="../publications/pdf/2023__Schmitt_et_al__AISTATS.pdf" class="btn btn-outline-primary btn-page-header btn-xs" role="button">PDF</a> <a href="https://proceedings.mlr.press/v206/schmitt23a.html" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Conference</a> <a href="http://arxiv.org/abs/2210.07278" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://github.com/marvinschmitt/MetaUncertaintyPaper" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Code &amp; Data</a> <a href="../talks/pdf/poster_meta_uncertainty.pdf" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Talk</a></p></li>
<li><p>Schmitt, M., Bürkner P. C., Köthe U., &amp; Radev S. T. (2023). Detecting Model Misspecification in Amortized Bayesian Inference with Neural Networks. <em>Proceedings of the German Conference on Pattern Recognition (GCPR)</em>. doi:10.1007/978-3-031-54605-1_35 <a href="../publications/pdf/2023__Schmitt_et_al__GCPR.pdf" class="btn btn-outline-primary btn-page-header btn-xs" role="button">PDF</a> <a href="https://link.springer.com/chapter/10.1007/978-3-031-54605-1_35" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Conference</a> <a href="https://arxiv.org/abs/2112.08866" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://github.com/marvinschmitt/ModelMisspecificationBF" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Code &amp; Data</a> <a href="../talks/pdf/detecting_MMS_bayesflow.pdf" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Talk</a> <a href="../talks/pdf/poster_detecting_MMS_bayesflow.pdf" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Poster</a></p></li>
<li><p>Radev S. T., Schmitt M., Pratz V., Picchini U., Köthe U., &amp; Bürkner P. C. (2023). JANA: Jointly Amortized Neural Approximation of Complex Bayesian Models. <em>Uncertainty in Artificial Intelligence (UAI) Conference Proceedings</em>. <a href="../publications/pdf/2023__Radev_et_al__UAI.pdf" class="btn btn-outline-primary btn-page-header btn-xs" role="button">PDF</a> <a href="https://proceedings.mlr.press/v216/radev23a.html" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Conference</a> <a href="https://arxiv.org/abs/2302.09125" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://github.com/bayesflow-org/JANA-Paper" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Code &amp; Data</a> <a href="../talks/pdf/glimpse_amortized_bayesian_inference.pdf" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Talk</a></p></li>
<li><p>Radev S. T., Schmitt M., Schumacher L., Elsemüller L., Pratz V., Schälte Y., Köthe U., &amp; Bürkner P. C. (2023). BayesFlow: Amortized Bayesian Workflows With Neural Networks. <em>Journal of Open Source Software</em>. doi:10.21105/joss.05702 <a href="../publications/pdf/2023__Radev_et_al__JOSS.pdf" class="btn btn-outline-primary btn-page-header btn-xs" role="button">PDF</a> <a href="https://joss.theoj.org/papers/10.21105/joss.05702" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Journal</a> <a href="https://arxiv.org/abs/2306.16015" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="../software#bayesflow" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Software</a></p></li>
</ul>
<p>Software:</p>
<ul>
<li>ggsimplex: Simplex visualizations with ggplot2 <a href="https://github.com/marvinschmitt/ggsimplex" class="btn btn-outline-primary btn-page-header btn-xs" role="button">GitHub</a></li>
</ul>
<!-- -->
</section>
<section id="joint-priors-mlms" class="level2">
<h2 class="anchored" data-anchor-id="joint-priors-mlms">Intuitive Joint Priors for Bayesian Multilevel Models</h2>
<p><img src="../images/coef_r2d2_marginal.png" class="img-fluid" alt="Marginal densities of the R2D2 prior's coefficients."></p>
<p>Regression models are ubiquitous in the quantitative sciences making up a big part of all statistical analysis performed on data. In the quantitative sciences, data often contains multilevel structure, for example, because of natural groupings of individuals or repeated measurement of the same individuals. Multilevel models (MLMs) are designed specifically to account for the nested structure in multilevel data and are a widely applied class of regression models. From a Bayesian perspective, the widespread success of MLMs can be explained by the fact that they impose joint priors over a set of parameters with shared hyper-parameters, rather than separate independent priors for each parameter. However, in almost all state-of-the-art approaches, different additive regression terms in MLMs, corresponding to different parameter sets, still receive mutually independent priors. As more and more terms are being added to the model while the number of observations remains constant, such models will overfit the data. This is highly problematic as it leads to unreliable or uninterpretable estimates, bad out-of-sample predictions, and inflated Type I error rates.</p>
<p>To solve these challenges, this project aims to develop, evaluate, implement, and apply intuitive joint priors for Bayesian MLMs. We hypothesize that our developed priors will enable the reliable and interpretable estimation of much more complex Bayesian MLMs than was previously possible.</p>
<p>Overarching Topics: <a href="../research#prior-specification" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Prior Specification</a> <a href="../research#uncertainty-quantification" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Uncertainty Quantification</a></p>
<p>Project Members: <a href="../people#javier-aguilar" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Javier Aguilar</a></p>
<p>Funders: <a href="https://www.dfg.de/en/index.jsp" class="btn btn-outline-primary btn-page-header btn-xs" role="button">German Research Foundation (DFG)</a> <a href="https://www.tu-dortmund.de/en/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">TU Dortmund University</a> <a href="https://www.uni-stuttgart.de/en/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">University of Stuttgart</a></p>
<p>Funding Period: 2021 – 2025</p>
<p>Publications:</p>
<ul>
<li><p>Aguilar J. E. &amp; Bürkner P. C. (in review). Generalized Decomposition Priors on R2. <em>ArXiv preprint</em>. <a href="https://arxiv.org/abs/2401.10180" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://osf.io/ns2cv/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Code &amp; Data</a></p></li>
<li><p>Reiser P., Aguilar J. E., Guthke A., &amp; Bürkner P. C. (in review). Uncertainty Quantification and Propagation in Surrogate-based Bayesian Inference. <em>ArXiv preprint</em>. <a href="https://arxiv.org/abs/2312.05153" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://github.com/philippreiser/bayesian-surrogate-uncertainty-paper" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Code &amp; Data</a></p></li>
<li><p>Aguilar J. E. &amp; Bürkner P. C. (2023). Intuitive Joint Priors for Bayesian Linear Multilevel Models: The R2D2M2 prior. <em>Electronic Journal of Statistics</em>. doi:10.1214/23-EJS2136 <a href="../publications/pdf/2023__Aguilar_Bürkner__Electronic_Journal_of_Statistics.pdf" class="btn btn-outline-primary btn-page-header btn-xs" role="button">PDF</a> <a href="https://doi.org/10.1214/23-EJS2136" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Journal</a> <a href="https://arxiv.org/abs/2208.07132" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://osf.io/wgsth/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Code &amp; Data</a> <a href="../talks/pdf/poster_R2D2M2_prior.pdf" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Talk</a></p></li>
</ul>
<!-- -->
</section>
<section id="ml4bmb" class="level2">
<h2 class="anchored" data-anchor-id="ml4bmb">Machine Learning for Bayesian Model Building</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/pad-model-taxonomy.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:80.0%" alt="An illustration of the PAD model taxonomy."></p>
</figure>
</div>
<p>The Bayesian approach to data analysis provides a consistent and flexible way to handle uncertainty in all observations, model parameters, and model structure using probability theory. However, building Bayesian models in a principled way remains a highly complex task requiring a lot of expertise and cognitive resources. In this project, we will develop a machine assisted workflow for building interpretable, robust, and well-predicting Bayesian models. Based on statistical theory, we will develop a framework for simulating realistic data with known modeling challenges. Subsequently, using neural network architectures tuned to the structure of the fitted Bayesian models, machines will be trained on the simulated data to provide automatic model evaluation and modeling recommendations that guide the user through the model building process using interactive visualizations. While leaving the modeling choices up to the user, the machine learns from the user’s decisions to improve its recommendations on the fly.</p>
<p>Overarching Topics: <a href="../research#machine-workflow" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Machine-Assisted Workflows</a> <a href="../research#model-comparison" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Model Comparison</a> <a href="../research#uncertainty-quantification" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Uncertainty Quantification</a></p>
<p>Project Members: <a href="../people#maximilian-scholz" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Maximilian Scholz</a></p>
<p>Funders: <a href="https://www.simtech.uni-stuttgart.de/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Cluster of Excellence SimTech</a></p>
<p>Funding Period: 2021 – 2024</p>
<p>Publications:</p>
<ul>
<li><p>Fazio L., Scholz M., &amp; Bürkner P. C. (in review). Generative Bayesian Modeling with Implicit Priors. <em>ArXiv preprint</em>. <a href="https://arxiv.org/abs/2408.06504" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://github.com/sims1253/implicit-priors" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Code &amp; Data</a></p></li>
<li><p>Scholz M. &amp; Bürkner P. C. (in review). Prediction can be safely used as a proxy for explanation in causally consistent Bayesian generalized linear models. <em>ArXiv preprint</em>. <a href="https://arxiv.org/abs/2210.06927" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://osf.io/xgkzv/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Code &amp; Data</a></p></li>
<li><p>Scholz M. &amp; Bürkner P. C. (in review). Posterior accuracy and calibration under misspecification in Bayesian generalized linear models. <em>ArXiv preprint</em>. <a href="https://arxiv.org/abs/2311.09081" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a> <a href="https://osf.io/tmdcf/" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Code &amp; Data</a></p></li>
<li><p>Bürkner P. C., Scholz M., &amp; Radev S. T. (2023). Some models are useful, but how do we know which ones? Towards a unified Bayesian model taxonomy. <em>Statistics Surveys</em>. doi:10.1214/23-SS145 <a href="../publications/pdf/2023__Bürkner_et_al__Statistics_Surveys.pdf" class="btn btn-outline-primary btn-page-header btn-xs" role="button">PDF</a> <a href="https://doi.org/10.1214/23-SS145" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Journal</a> <a href="http://arxiv.org/abs/2209.02439" class="btn btn-outline-primary btn-page-header btn-xs" role="button">Preprint</a></p></li>
</ul>
<p>Software:</p>
<ul>
<li>bayesim: Simulations for Bayesian models <a href="https://github.com/sims1253/bayesim" class="btn btn-outline-primary btn-page-header btn-xs" role="button">GitHub</a></li>
<li>bayeshear: Metrics for evaluating Bayesian models <a href="https://github.com/sims1253/bayeshear" class="btn btn-outline-primary btn-page-header btn-xs" role="button">GitHub</a></li>
<li>bayesfam: Custom families for brms <a href="https://github.com/sims1253/bayesfam" class="btn btn-outline-primary btn-page-header btn-xs" role="button">GitHub</a></li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>Copyright 2024, Paul Bürkner</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item">
    <a class="nav-link" href="../impressum/index.html">
<p>Impressum</p>
</a>
  </li>  
</ul>
    </div>
  </div>
</footer>




</body></html>