<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Paul Bürkner - Research Projects</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="../site_libs/quarto-contrib/fontawesome6-0.1.0/all.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/fontawesome6-0.1.0/latex-fontsize.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Paul Bürkner</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../research/index.html" rel="" target="">
 <span class="menu-text">Research</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../people/index.html" rel="" target="">
 <span class="menu-text">People</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../projects/index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Projects</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../publications/index.html" rel="" target="">
 <span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../software/index.html" rel="" target="">
 <span class="menu-text">Software</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../talks/index.html" rel="" target="">
 <span class="menu-text">Talks</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../positions/index.html" rel="" target="">
 <span class="menu-text">Positions</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../cv/index.html" rel="" target="">
 <span class="menu-text">CV</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="mailto:paul.buerkner@gmail.com" rel="" target="">
 <span class="menu-text"><i class="fa-solid fa-envelope" aria-label="envelope"></i></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/paul-buerkner" rel="" target=""><i class="bi bi-github" role="img" aria-label="github">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://scholar.google.com/citations?user=JSj6m1IAAAAJ&amp;hl" rel="" target="">
 <span class="menu-text"><i class="fa-solid fa-graduation-cap" aria-label="graduation-cap"></i></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/paulbuerkner" rel="" target=""><i class="bi bi-twitter" role="img" aria-label="twitter">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
        
    </div>
<!-- main -->
<main class="content column-page" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Research Projects</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="bayesflow-software" class="level2">
<h2 class="anchored" data-anchor-id="bayesflow-software">BayesFlow: Simulation Intelligence with Deep Learning</h2>
<p>Simulation intelligence (SI) subsumes an emerging generation of scientific methods which utilize digital simulations for emulating and understanding complex real-world systems and phenomena. Recently, neural networks and deep learning have demonstrated a great potential for accelerating and scaling up SI to previously intractable problems and data sets. However, the availability of user-friendly software is still limited, which hampers the widespread and flexible use of modern SI methods.</p>
<p>In this project, we focus on software for amortized Bayesian inference, which is an essential part of SI. The hallmark feature of amortized Bayesian inference is an upfront training phase (e.g., of a neural network), which is then amortized by a nearly instant fully Bayesian inference for an arbitrary number of data sets during test time. Concretely, we aim to advance the <a href="../software#bayesflow">BayesFlow research software library</a> into becoming the long-term, gold-standard software for amortized Bayesian inference.</p>
<p>Overarching Topics: <a href="../research#sbi">Simulation-Based Inference</a>, <a href="../research#machine-workflow">Machine-Assisted Bayesian Workflow</a></p>
<p>Project Members: <a href="../people#lars-kuehmichel">Lars Kühmichel</a></p>
<p>Funders: <a href="https://www.dfg.de/en/index.jsp">German Research Foundation (DFG)</a></p>
<p>Funding Period: 2024 – 2027</p>
</section>
<section id="abi-applications" class="level2">
<h2 class="anchored" data-anchor-id="abi-applications">Applications of Amortized Bayesian Inference</h2>
<p>Recent developments in simulation-based amortized inference have ushered in new possibilities for conducting principled Bayesian analysis. The simulation-based approach unlocks the potential of complex models whose likelihoods or priors are not analytically tractable. Amortized approaches make the required computations relatively fast, thus allowing for the deployment of intricate models in scenarios that were hitherto deemed unfeasible or inconvenient. Nevertheless, the novelty of this approach poses a challenge, as its widespread adoption hinges on the availability of user-friendly documentation and resources that simplify entry into the field, as well as empirical examples that validate the method’s usefulness for the practical researchers.</p>
<p>In this project, our emphasis is on applications within cognitive modeling and developmental psychology. We focus on how simulation-based amortized inference can address important challenges within the field, not only during the data analysis phase but also in the planning and execution of studies and experiments. As a by-product we will generate tutorials and educational materials providing gentle introductions into the topic. This project also aims to lay the foundations for integrating simulation-based amortized inference with popular statistical software packages used by practitioners who may not have extensive coding skills, thereby broadening the scope of users benefiting from its advantages.</p>
<p>Overarching Topics: <a href="../research#sbi">Simulation-Based Inference</a>, <a href="../research#uncertainty-quantification">Uncertainty Quantification</a></p>
<p>Project Members: <a href="../people#simon-kucharsky">Šimon Kucharský</a></p>
<p>Funders: <a href="https://www.tu-dortmund.de/en/">TU Dortmund University</a></p>
<p>Funding Period: 2024 – 2027</p>
</section>
<section id="amortized-mlms" class="level2">
<h2 class="anchored" data-anchor-id="amortized-mlms">Amortized Bayesian Inference for Multilevel Models</h2>
<p>Probabilistic multilevel models (MLMs) are a central building block in Bayesian data analysis. Despite their widely acknowledged advantages, MLMs remain challenging to estimate and evaluate, especially when the involved likelihoods or priors are analytically intractable. Recent developments in generative deep learning and simulation-based inference have shown promising results in scaling up Bayesian inference through amortization. However, the utility of deep generative models for learning Bayesian MLMs remains largely unexplored.</p>
<p>In this project, we propose to develop a general and efficient neural inference framework for estimating and evaluating complex Bayesian MLMs. Our framework will substantially extend previous work on simulation-based Bayesian inference for single-level models. Moreover, it aims to encompass not only the inference phase of a Bayesian workflow but also the model evaluation steps, which usually comprise a computational bottleneck with standard (non-amortized) Bayesian methods. Thus, the proposed project has the potential to greatly enhance model-based inference and understanding of complex processes across the quantitative sciences.</p>
<p>Overarching Topics: <a href="../research#sbi">Simulation-Based Inference</a>, <a href="../research#uncertainty-quantification">Uncertainty Quantification</a></p>
<p>Project Members: <a href="../people#daniel-habermann">Daniel Habermann</a></p>
<p>Funders: <a href="https://www.dfg.de/en/index.jsp">German Research Foundation (DFG)</a></p>
<p>Funding Period: 2023 – 2026</p>
<p>Publications:</p>
<ul>
<li>Schmitt M., Habermann D., Bürkner P. C., Köthe U., &amp; Radev S. T. (2023). <a href="https://arxiv.org/abs/2310.04395">Leveraging Self-Consistency for Data-Efficient Amortized Bayesian Inference</a>. <em>ArXiv preprint</em>.</li>
</ul>
</section>
<section id="bdlvms" class="level2">
<h2 class="anchored" data-anchor-id="bdlvms">Bayesian Distributional Latent Variable Models</h2>
<p>In psychology and related sciences, a lot of research is concerned with studying latent variables, that is, constructs which are not directly observable. Statistical methods for modeling latent variables based on manifest (observable) indicators are thus crucial to the scientific progress in those fields. Two major interconnected statistical areas dealing with latent variables exist, namely, Item Response Theory (IRT) and Structural Equation Modeling (SEM). Although the two fields are closely connected, the frontiers of IRT and SEM have developed in somewhat different directions.</p>
<p>A combination of these two major frontiers would enable researchers to tackle a lot of advanced psychological research questions at the intersection of psychometrics, personnel psychology, cognitive psychology, and applied psychology. In order for us to gain better insights into behavioral and cognitive processes, their mathematical approximations should match the processes’ complexity in both overall distributional form and its components that are expressed as complex functions of predicting variables.</p>
<p>This project aims to develop a framework for Bayesian distributional latent variable models that combines the principles of IRT and SEM with the flexibility of distributional regression powered by modern Bayesian estimation methods.</p>
<p>Overarching Topics: <a href="../research#uncertainty-quantification">Uncertainty Quantification</a>, <a href="../research#machine-workflow">Machine-Assisted Bayesian Workflow</a></p>
<p>Project Members: <a href="../people#luna-fazio">Luna Fazio</a></p>
<p>Funders: <a href="https://www.dfg.de/en/index.jsp">German Research Foundation (DFG)</a>, <a href="https://www.tu-dortmund.de/en/">TU Dortmund University</a></p>
<p>Funding Period: 2022 – 2025</p>
<!-- -->
</section>
<section id="sbpriors" class="level2">
<h2 class="anchored" data-anchor-id="sbpriors">Simulation-Based Prior Distributions for Bayesian Models</h2>
<div class="quarto-figure quarto-figure-left">
<figure class="figure">
<p><img src="../images/simulation-based-priors.png" class="img-fluid figure-img" alt="An illustration of the simulation-based prior workflow."></p>
</figure>
</div>
<p>Data-driven statistical modeling plays a crucial role in almost all quantitative sciences. Despite continuous increases in the amount of available data, the addition of further information sources, such as expert knowledge, often remains an irreplaceable part of setting up high-fidelity models. Grounded in probability theory, Bayesian statistics provides a principled approach to including expert knowledge in the form of prior distributions, a process called prior elicitation. However, prior elicitation for high-dimensional Bayesian models is infeasible with existing methods due to practical and computational challenges. With the goal of solving these challenges, we propose to develop simulation-based priors for high-dimensional Bayesian models that allow to incorporate prior information elicited on any model-implied quantities. We expect the developed methods to have a major impact on all fields applying probabilistic modeling by making the use of expert knowledge practical, robust, and computationally feasible.</p>
<p>Overarching Topics: <a href="../research#prior-specification">Prior Specification</a>, <a href="../research#sbi">Simulation-Based Inference</a></p>
<p>Project Members: <a href="../people#florence-bockting">Florence Bockting</a></p>
<p>Funders: <a href="https://www.tu-dortmund.de/en/">TU Dortmund University</a>, <a href="https://www.simtech.uni-stuttgart.de/">Cluster of Excellence SimTech</a></p>
<p>Funding Period: 2022 – 2025</p>
<p>Publications:</p>
<ul>
<li>Bockting F., Radev, S. T., &amp; Bürkner P. C. (2023). <a href="http://arxiv.org/abs/2308.11672">Simulation-Based Prior Knowledge Elicitation for Parametric Bayesian Models</a>. <em>ArXiv preprint</em>.</li>
</ul>
<!-- -->
</section>
<section id="scrna-models" class="level2">
<h2 class="anchored" data-anchor-id="scrna-models">Probabilistic Models for Single-Cell RNA Sequencing Data</h2>
<div class="quarto-figure quarto-figure-left">
<figure class="figure">
<p><img src="../images/latent-derivative-gps.png" class="img-fluid figure-img" alt="An illustration of our latent derivative Gaussian process framework."></p>
</figure>
</div>
<p>Trajectory and pseudo-time inference methods in single-cell RNA sequencing face challenges from the ambiguity of the static single-cell transcriptome snapshot data. In this project, we aim to tackle this challenge by means of advanced probabilistic methods. Concretely, we aim to reconstruct unobserved cell ordering as latent pseudo-time by analyzing RNA spliced counts and corresponding derivative RNA velocity. Further, we aim to obtain uncertainty estimates of the latent cell ordering using Bayesian inference. To achieve these goals, we will develop advanced latent Gaussian process models with the ability of utilizing derivative information to increase precision in estimating unobserved latent inputs. This model deploys derivative covariance kernel functions and modifications in the hyperparameter specifications, thus increasing capabilities for utilizing derivative information in a multi-output setting. Although the primary motivation lies in applications in single-cell biology, this model has the potential to solve similar research problems dealing with multi-output data and its derivatives from diverse fields of study.</p>
<p>Overarching Topics: <a href="../research#uncertainty-quantification">Uncertainty Quantification</a></p>
<p>Project Members: <a href="../people#soham-mukherjee">Soham Mukherjee</a></p>
<p>Co-Supervisors: <a href="https://claassenlab.github.io/people/manfred-claassen/">Manfred Claassen</a></p>
<p>Funders: <a href="https://www.tu-dortmund.de/en/">TU Dortmund University</a>, <a href="https://uni-tuebingen.de/en/">University of Tübingen</a></p>
<p>Funding Period: 2022 – 2025</p>
<!-- -->
</section>
<section id="uncertainty-surrogate-models" class="level2">
<h2 class="anchored" data-anchor-id="uncertainty-surrogate-models">Data-Integrated Training of Surrogate Models for Uncertainty Quantification</h2>
<p><img src="../images/uncertainty_prop_workflow.png" class="img-fluid" alt="Surrogate uncertainty propagation workflow."></p>
<p>Uncertainty quantification is crucial to assess the predictive power and limitations of complex systems models. However, in the case of high-dimensional parameter spaces and/or complex functional relationships, physics-based simulation models are often computationally too demanding for rigorous Bayesian uncertainty quantification. Surrogate models allow for such analyses with much lower effort. They are typically trained such that they fit the simulation reference best.</p>
<p>What is left unexplored is the possibility of surrogate models to actually fit observed data better than the reference model. This phenomenon occurs when structural misspecification of the physics-constrained reference model limits its performance, but at the same time, the more flexible data-driven surrogate model can better capture the relation of output and input data. Such situations offer huge potential for diagnostic evaluation of the modelling approach toward deeper system understanding and model improvement.</p>
<p>We aim at developing (1) a weighted data-integrated surrogate training approach for improved predictive performance, (2) a diagnostic approach for structural error detection in the reference model, and (3) an uncertainty propagation analysis that accounts for the approximation error introduced by the use of surrogates.</p>
<p>Overarching Topics: <a href="../research#uncertainty-quantification">Uncertainty Quantification</a></p>
<p>Project Members: <a href="../people#philipp-reiser">Philipp Reiser</a></p>
<p>Co-Supervisors: <a href="https://www.simtech.uni-stuttgart.de/exc/people/Guthke/">Anneli Guthke</a></p>
<p>Funders: <a href="https://www.simtech.uni-stuttgart.de/">Cluster of Excellence SimTech</a></p>
<p>Funding Period: 2022 – 2025</p>
<p>Publications:</p>
<ul>
<li>Reiser, P., Aguilar J. E., Guthke A., &amp; Bürkner P. C. (2023). <a href="https://arxiv.org/abs/2312.05153">Uncertainty Quantification and Propagation in Surrogate-based Bayesian Inference</a>. <em>ArXiv preprint</em>.</li>
</ul>
<!-- -->
</section>
<section id="mu-bmb" class="level2">
<h2 class="anchored" data-anchor-id="mu-bmb">Meta-Uncertainty in Bayesian Model Comparison</h2>
<p><img src="../images/meta_uncertainty_banner.png" class="img-fluid" alt="An illustration of meta-uncertainty."></p>
<p>What we can learn from a single data set in experiments and observational studies is always limited, and we are inevitably left with some remaining uncertainty. It is of utmost importance to take this uncertainty into account when drawing conclusions if we want to make real scientific progress. Formalizing and quantifying uncertainty is thus at the heart of statistical methods aiming to obtain insights from data.</p>
<p>To compare scientific theories, scientists translate them into statistical models and then investigate how well the models’ predictions match the gathered real-world data. One widely applied approach to compare statistical models is Bayesian model comparison (BMC). Relying on BMC, researchers obtain the probability that each of the competing models is true (or is closest to the truth) given the data. These probabilities are measures of uncertainty and, yet, are also uncertain themselves. This is what we call meta-uncertainty (uncertainty over uncertainties). Meta-uncertainty affects the conclusions we can draw from model comparisons and, consequently, the conclusions we can draw about the underlying scientific theories.</p>
<p>This project contributes to this endeavor by developing and evaluating methods for quantifying meta-uncertainty in BMC. Building upon mathematical theory of meta-uncertainty, we will utilize extensive model simulations as an additional source of information, which enable us to quantify so-far implicit yet important assumptions of BMC. What is more, we will be able to differentiate between a closed world, where the true model is assumed to be within the set of considered models, and an open world, where the true model may not be within that set – a critical distinction in the context of model comparison procedures.</p>
<p>Overarching Topics: <a href="../research#model-comparison">Model Comparison</a>, <a href="../research#uncertainty-quantification">Uncertainty Quantification</a>, <a href="../research#sbi">Simulation-Based Inference</a></p>
<p>Project Members: <a href="../people#marvin-schmitt">Marvin Schmitt</a></p>
<p>Funders: <a href="https://cyber-valley.de/en/research-fund">Cyber Valley Research Fund</a></p>
<p>Funding Period: 2021 – 2024</p>
<p>Publications:</p>
<ul>
<li>Schmitt, M., Radev, S. T., &amp; Bürkner P. C. (2023). <a href="http://arxiv.org/abs/2210.07278">Meta-Uncertainty in Bayesian Model Comparison</a>. <em>Artificial Intelligence and Statistics (AISTATS) Conference Proceedings</em>.</li>
<li>Schmitt, M., Bürkner P. C., Köthe U., &amp; Radev S. T. (2023). <a href="https://arxiv.org/abs/2112.08866">Detecting Model Misspecification in Amortized Bayesian Inference with Neural Networks</a>. <em>Proceedings of the German Conference on Pattern Recognition (GCPR)</em>.</li>
<li>Schmitt, M., Radev, S. T., &amp; Bürkner P. C. (2023). <a href="https://arxiv.org/abs/2311.10671">Fuse It or Lose It: Deep Fusion for Multimodal Simulation-Based Inference</a>. <em>ArXiv preprint</em>.</li>
<li>Schmitt M., Pratz V., Köthe U., Bürkner P. C., &amp; Radev S. T. (2023). <a href="https://arxiv.org/abs/2312.05440">Consistency Models for Scalable and Fast Simulation-Based Inference</a>. <em>ArXiv preprint</em>.</li>
<li>Schmitt M., Habermann D., Bürkner P. C., Köthe U., &amp; Radev S. T. (2023). <a href="https://arxiv.org/abs/2310.04395">Leveraging Self-Consistency for Data-Efficient Amortized Bayesian Inference</a>. <em>ArXiv preprint</em>.</li>
<li>Radev S. T., Schmitt M., Pratz V., Picchini U., Köthe U., &amp; Bürkner P. C. (2023). <a href="https://arxiv.org/abs/2302.09125">JANA: Jointly Amortized Neural Approximation of Complex Bayesian Models</a>. <em>Uncertainty in Artificial Intelligence (UAI) Conference Proceedings</em>.</li>
</ul>
<p>Software:</p>
<ul>
<li>ggsimplex: Simplex visualizations with ggplot2 [<a href="https://github.com/marvinschmitt/ggsimplex">GitHub</a>]</li>
</ul>
<!-- -->
</section>
<section id="joint-priors-mlms" class="level2">
<h2 class="anchored" data-anchor-id="joint-priors-mlms">Intuitive Joint Priors for Bayesian Multilevel Models</h2>
<p><img src="../images/coef_r2d2_marginal.png" class="img-fluid" alt="Marginal densities of the R2D2 prior's coefficients."></p>
<p>Regression models are ubiquitous in the quantitative sciences making up a big part of all statistical analysis performed on data. In the quantitative sciences, data often contains multilevel structure, for example, because of natural groupings of individuals or repeated measurement of the same individuals. Multilevel models (MLMs) are designed specifically to account for the nested structure in multilevel data and are a widely applied class of regression models. From a Bayesian perspective, the widespread success of MLMs can be explained by the fact that they impose joint priors over a set of parameters with shared hyper-parameters, rather than separate independent priors for each parameter. However, in almost all state-of-the-art approaches, different additive regression terms in MLMs, corresponding to different parameter sets, still receive mutually independent priors. As more and more terms are being added to the model while the number of observations remains constant, such models will overfit the data. This is highly problematic as it leads to unreliable or uninterpretable estimates, bad out-of-sample predictions, and inflated Type I error rates.</p>
<p>To solve these challenges, this project aims to develop, evaluate, implement, and apply intuitive joint priors for Bayesian MLMs. We hypothesize that our developed priors will enable the reliable and interpretable estimation of much more complex Bayesian MLMs than was previously possible.</p>
<p>Overarching Topics: <a href="../research#prior-specification">Prior Specification</a></p>
<p>Project Members: <a href="../people#javier-aguilar">Javier Aguilar</a></p>
<p>Funders: <a href="https://www.dfg.de/en/index.jsp">German Research Foundation (DFG)</a>, <a href="https://www.tu-dortmund.de/en/">TU Dortmund University</a>, <a href="https://www.uni-stuttgart.de/en/">University of Stuttgart</a></p>
<p>Funding Period: 2021 – 2025</p>
<p>Publications:</p>
<ul>
<li>Aguilar J. E. &amp; Bürkner P. C. (2024). <a href="https://arxiv.org/abs/2401.10180">Generalized Decomposition Priors on R2</a>. <em>ArXiv preprint</em>.</li>
<li>Aguilar J. E. &amp; Bürkner P. C. (2023). <a href="https://doi.org/10.1214/23-EJS2136">Intuitive Joint Priors for Bayesian Linear Multilevel Models: The R2D2M2 prior</a>. <em>Electronic Journal of Statistics</em>. Available for free also on <a href="https://arxiv.org/abs/2208.07132">ArXiv</a>. doi:10.1214/23-EJS2136</li>
<li>Reiser, P., Aguilar J. E., Guthke A., &amp; Bürkner P. C. (2023). <a href="https://arxiv.org/abs/2312.05153">Uncertainty Quantification and Propagation in Surrogate-based Bayesian Inference</a>. <em>ArXiv preprint</em>.</li>
</ul>
<!-- -->
</section>
<section id="ml4bmb" class="level2">
<h2 class="anchored" data-anchor-id="ml4bmb">Machine Learning for Bayesian Model Building</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/pad-model-taxonomy.png" class="img-fluid figure-img" style="width:80.0%" alt="An illustration of the PAD model taxonomy."></p>
</figure>
</div>
<p>The Bayesian approach to data analysis provides a consistent and flexible way to handle uncertainty in all observations, model parameters, and model structure using probability theory. However, building Bayesian models in a principled way remains a highly complex task requiring a lot of expertise and cognitive resources. In this project, we will develop a machine assisted workflow for building interpretable, robust, and well-predicting Bayesian models. Based on statistical theory, we will develop a framework for simulating realistic data with known modeling challenges. Subsequently, using neural network architectures tuned to the structure of the fitted Bayesian models, machines will be trained on the simulated data to provide automatic model evaluation and modeling recommendations that guide the user through the model building process using interactive visualizations. While leaving the modeling choices up to the user, the machine learns from the user’s decisions to improve its recommendations on the fly.</p>
<p>Overarching Topics: <a href="../research#machine-workflow">Machine-Assisted Bayesian Workflow</a>, <a href="../research#model-comparison">Model Comparison</a></p>
<p>Project Members: <a href="../people#maximilian-scholz">Maximilian Scholz</a></p>
<p>Funders: <a href="https://www.simtech.uni-stuttgart.de/">Cluster of Excellence SimTech</a></p>
<p>Funding Period: 2021 – 2024</p>
<p>Publications:</p>
<ul>
<li>Scholz M. &amp; Bürkner P. C. (2023). <a href="https://arxiv.org/abs/2210.06927">Prediction can be safely used as a proxy for explanation in causally consistent Bayesian generalized linear models</a>. <em>ArXiv preprint</em>.</li>
<li>Scholz M. &amp; Bürkner P. C. (2023). <a href="https://arxiv.org/abs/2311.09081">Posterior accuracy and calibration under misspecification in Bayesian generalized linear models</a>. <em>ArXiv preprint</em>.</li>
<li>Bürkner P. C., Scholz M., &amp; Radev S. T. (2023). <a href="http://arxiv.org/abs/2209.02439">Some models are useful, but how do we know which ones? Towards a unified Bayesian model taxonomy</a>. <em>Statistics Surveys</em>.</li>
</ul>
<p>Software:</p>
<ul>
<li>bayesim: Simulations for Bayesian models [<a href="https://github.com/sims1253/bayesim">GitHub</a>]</li>
<li>bayeshear: Metrics for evaluating Bayesian models [<a href="https://github.com/sims1253/bayeshear">GitHub</a>]</li>
<li>bayesfam: Custom families for brms [<a href="https://github.com/sims1253/bayesfam">GitHub</a>]</li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">Copyright 2024, Paul Bürkner</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item">
    <a class="nav-link" href="../impressum/index.html">Impressum</a>
  </li>  
</ul>
    </div>
  </div>
</footer>



</body></html>